src/Common/src/System/Security/Cryptography/DSACng.ImportExport.cs

```diff
@@ -0,0 +1,377 @@
+                        else
+                        {
+                            Debug.Assert(magic == KeyBlobMagicNumber.BCRYPT_DSA_PUBLIC_MAGIC_V2 || magic == KeyBlobMagicNumber.BCRYPT_DSA_PRIVATE_MAGIC_V2);
+
+                            if (dsaBlob.Length < sizeof(BCRYPT_DSA_KEY_BLOB_V2))
+                                throw ErrorCode.E_FAIL.ToCryptographicException();
+
+                            BCRYPT_DSA_KEY_BLOB_V2* pBcryptBlob = (BCRYPT_DSA_KEY_BLOB_V2*)pDsaBlob;
+                            // We now have a buffer laid out as follows:
+                            //     BCRYPT_DSA_KEY_BLOB_V2  header
+                            //     byte[cbSeedLength]      Seed
+                            //     byte[cbGroupSize]       Q
+                            //     byte[cbKey]             P
+                            //     byte[cbKey]             G
+                            //     byte[cbKey]             Y
+                            //     -- Private only --
+                            //     byte[cbGroupSize]       X
+
+                            offset = sizeof(BCRYPT_DSA_KEY_BLOB_V2) - 4; //skip to Count[4]
+
+                            // Read out a BCRYPT_DSA_KEY_BLOB_V2 structure.
+                            dsaParams.Counter = FromBigEndian(Interop.BCrypt.Consume(dsaBlob, ref offset, 4));
+
+                            // We better have computed the right allocation size above!
+                            Debug.Assert(offset == sizeof(BCRYPT_DSA_KEY_BLOB_V2), "offset == sizeof(BCRYPT_DSA_KEY_BLOB_V2)");
+
+                            dsaParams.Seed = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbSeedLength);
+                            dsaParams.Q = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbGroupSize);
+                            dsaParams.P = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbKey);
+                            dsaParams.G = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbKey);
+                            dsaParams.Y = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbKey);
+                            if (includePrivateParameters)
+                            {
+                                dsaParams.X = Interop.BCrypt.Consume(dsaBlob, ref offset, pBcryptBlob->cbGroupSize);
+                            }
+                        }
+
+                        // If no counter/seed information is present, normalize Counter and Seed to 0/null to maintain parity with the CAPI version of DSA.
+                        if (dsaParams.Counter == -1)
+                        {
+                            dsaParams.Counter = 0;
+                            dsaParams.Seed = null;
+                        }
+
+                        Debug.Assert(offset == 0, "offset == 0");
+
+                        return dsaParams;
+                    }
+                }
+            }
+
+            private static byte[] ToBigEndian(int i)
```

Comment:
Makes sense. Note this method was ported from desktop CLR. I changed to
        internal static void EmitBigEndian(byte[] blob, ref int offset, int value)
        {
            blob[offset + 0] = ((byte)(value >> 24));
            blob[offset + 1] = ((byte)(value >> 16));
            blob[offset + 2] = ((byte)(value >> 8));
            blob[offset + 3] = ((byte)(value));
            offset += 4;
        }